llvm/tools/llvm-exegesis/llvm-exegesis.cpp

```diff
@@ -509,22 +509,42 @@ void benchmarkMain() {
   }
 
   const auto Opcodes = getOpcodesOrDie(State);
+  std::vector<BenchmarkCode> Configurations;
+
+  unsigned LoopRegister =
+      State.getExegesisTarget().getDefaultLoopCounterRegister(
+          State.getTargetMachine().getTargetTriple());
+
+  if (Opcodes.empty()) {
```

Comment:
Why move this code ?

Reply:
With the new setup, each call to `SnippetRepetitor::Create` needs to have the loop register, but `generateSnippets` relies on `SnippetRepetitor::getReservedRegs`, so I've moved the snippet setup code above so we can get the loop register to initialize the loop register for creating the repetitor and left the snippet generation code below so that everything is defined before it is used.